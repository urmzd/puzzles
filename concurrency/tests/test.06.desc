##
## This test tests the queue implementation
## between the reader thread and the main
## thread
## 
## Input: A series of TRX events, followed by a
##        BLK event and an END event with a 
##        delay in the input at the BLK event.
##        this causes the main thread to run 
##        and use up the input.
## 
## Part 1: Basic functionality
## The output of your program with the 
## "Thread..." and "Received ..." statesments
## stripped out should match the solution's
## output with "Thread..." and "Received ..." 
## statesments stripped out.
## 
## Part 2: Receive ordering
## The order of all receive statements generated
## by your program should be the same as in the 
## solution.
##
