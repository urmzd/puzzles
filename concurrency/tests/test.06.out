Received event TRX with ID 11235
Received event TRX with ID 11236
Received event TRX with ID 11237
Received event TRX with ID 11238
Received event TRX with ID 11239
Received event TRX with ID 11240
Received event TRX with ID 11241
Received event TRX with ID 11242
Received event TRX with ID 11243
Adding transaction: 11235 Al_Alpha Beth_Beta 150000 40
Adding transaction: 11236 Cat_Charlie Dov_Delta 150000 20
Adding transaction: 11237 Eve_Echo Fin_Foxtrot 150000 25
Adding transaction: 11238 Guy_Golf Hana_Hotel 150000 40
Adding transaction: 11239 Iona_Indigo Jeff_Juliet 150000 20
Adding transaction: 11240 Kit_Kilo Liam_Lima 150000 25
Adding transaction: 11241 Mona_Mike Nick_November 150000 40
Adding transaction: 11242 Otto_Oscar Peta_Papa 150000 20
Adding transaction: 11243 Qu_Quebec Reya_Romeo 150000 25
Received event BLK with ID 1
Received event Al_Alpha
Received event Beth_Beta
Received event 150000
Received event 40
Received event 11238
Received event Guy_Golf
Received event Hana_Hotel
Received event 150000
Received event 40
Received event 11241
Received event Mona_Mike
Received event Nick_November
Received event 150000
Received event 40
Received event 11236
Received event Cat_Charlie
Received event Dov_Delta
Received event 150000
Received event 20
Received event 11237
Received event Eve_Echo
Received event Fin_Foxtrot
Received event 150000
Received event 25
Received event 11239
Received event Iona_Indigo
Received event Jeff_Juliet
Received event 150000
Received event 20
Received event 0x00000044
Received event 0x0031353d
Received event END
